### PR TITLE
Allow injection of SearchSession/SearchMapping

### PR DESCRIPTION
**Pull request by yrodiere**
~~Based on #13694, which should be merged first. Creating this PR as a draft until it's done.~~

Related quickstart update: https://github.com/quarkusio/quarkus-quickstarts/pull/725

### CODE REVIEW

File: extensions/hibernate-search-orm-elasticsearch/deployment/pom.xml [hunk 1]
```diff
@@ -52,6 +67,118 @@
                     </annotationProcessorPaths>
                 </configuration>
             </plugin>
+            <plugin>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <skip>true</skip>
```
Comment (gastaldi) [inline extensions/hibernate-search-orm-elasticsearch/deployment/pom.xml:73]: Is this intended?

File: extensions/hibernate-search-orm-elasticsearch/deployment/src/test/resources/application.properties [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+quarkus.datasource.jdbc.url=jdbc:h2:mem:default;DB_CLOSE_DELAY=-1
+
+quarkus.hibernate-orm.dialect=org.hibernate.dialect.H2Dialect
+quarkus.hibernate-orm.database.generation=drop-and-create
+
+quarkus.hibernate-search-orm.elasticsearch.version=7
+quarkus.hibernate-search-orm.elasticsearch.hosts=${elasticsearch.hosts:localhost:9200}
+quarkus.hibernate-search-orm.elasticsearch.protocol=${elasticsearch.protocol:http}
+quarkus.hibernate-search-orm.schema-management.strategy=drop-and-create-and-drop
+quarkus.hibernate-search-orm.automatic-indexing.synchronization.strategy=sync
```
Comment (gsmet) [inline extensions/hibernate-search-orm-elasticsearch/deployment/src/test/resources/application.properties:11]: Don't you expect the configuration to be potentially different depending on the PU?

Doesn't need to be part of this PR, just asking.

File: extensions/hibernate-search-orm-elasticsearch/deployment/pom.xml (hunk 1 above)
Comment (gsmet) [reply extensions/hibernate-search-orm-elasticsearch/deployment/pom.xml:73]: Yes, they need Elasticsearch so they are run if `-Dtest-containers`

File: extensions/hibernate-search-orm-elasticsearch/deployment/src/test/resources/application.properties (hunk 2 above)
Comment (yrodiere) [reply extensions/hibernate-search-orm-elasticsearch/deployment/src/test/resources/application.properties:11]: Right now you can make do by defining one backend per PU, but some settings (like this one) will have to be the same on all PUs, which is not great.

So yes, we'd need additional changes to fully support multiple PUs. In another PR :)

File: docs/src/main/asciidoc/hibernate-search-orm-elasticsearch.adoc [hunk 3]
```diff
@@ -14,12 +14,6 @@ We will also explore how you can can query your Elasticsearch cluster using the
 
 include::./status-include.adoc[]
 
-[WARNING]
-====
-This extension is based on a beta version of Hibernate Search.
-While APIs are quite stable and the code is of production quality and thoroughly tested, some features are still missing, performance might not be optimal and some APIs or configuration properties might change as the extension matures.
-====
-
```
Comment (Sanne) [inline docs/src/main/asciidoc/hibernate-search-orm-elasticsearch.adoc:22]: :100:

File: docs/src/main/asciidoc/hibernate-search-orm-elasticsearch.adoc [hunk 4]
```diff
@@ -481,16 +475,26 @@ It is definitely too aggressive for person names but it is perfect for the book
...
+In our existing `LibraryResource`, we just need to inject the `SearchSession`:
+
+[source,java]
+----
+    @Inject
+    SearchSession searchSession; // <1>
+----
+<1> Inject a Hibernate Search session, which relies on the `EntityManager` under the hood.
+Applications with multiple persistence units can use a CDI qualifier to select the right one:
+either `@Named` or `@io.quarkus.hibernate.orm.PersistenceUnit`.
```
Comment (Sanne) [inline docs/src/main/asciidoc/hibernate-search-orm-elasticsearch.adoc:487]: Not a blocker, but I'd prefer such a note to be fully expressed with an example.

**Review by Sanne — APPROVED** [review]

File: docs/src/main/asciidoc/hibernate-search-orm-elasticsearch.adoc (hunk 4 above)
Comment (yrodiere) [reply docs/src/main/asciidoc/hibernate-search-orm-elasticsearch.adoc:487]: Ok, I'll add an example in the next PR that adds first-class support for multiple PUs.

**Pull request merged**